### PR TITLE
Fixer for Prettier present

### PR DESCRIPTION
**Pull request by vlasy**
## Description

Add fixer for JavaScript.PrettierUsed practice.
It will:
- install prettier
- if not present, create prettier config with default prettier values
- if not present, add "format" npm script to package.json which runs the prettier

## Caveats
- default values in config are from prettier of DXS, not prettier which is installed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## TODO:
PackageManagerUtils code looks like it should be part of JavascriptPackageInspector - I want to move it there right after this MR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,6 +99,7 @@
     "jest-mock-extended": "1.0.8",
     "nock": "12.0.3",
     "prettier": "1.19.1",
+    "prettier-default-config": "^0.2.3",
```
Comment (ryzzaki) [inline package.json:102]:
```suggestion
    "prettier-default-config": "0.2.3",
```
I think it's better to specify the exact version of the package :D you might get some lock file issues after changing it, so I suggest running `yarn install` post-change :)

File: src/practices/JavaScript/PrettierUsedPractice.spec.ts [hunk 2]
```diff
@@ -54,4 +61,44 @@ describe('PrettierUsedPractice', () => {
     const result = await practice.isApplicable(containerCtx.practiceContext);
     expect(result).toEqual(false);
   });
+
+  describe('Fixer', () => {
+    afterEach(async () => {
+      jest.clearAllMocks();
```
Comment (ryzzaki) [inline src/practices/JavaScript/PrettierUsedPractice.spec.ts:67]:
```suggestion
      jest.clearAllMocks();
      containerCtx.virtualFileSystemService.clearFileSystem();
```

File: package.json (hunk 1 above)
Comment (prokopsimek) [reply package.json:102]: It's definitely better because when the user installs the DX Scanner from NPM globally so it doesn't respect a lockfile.

**Pull request merged**